### ISSUE TITLE
Budgie Settings not working

### ISSUE THREAD

**Issue by Vistaus** [budgie-desktop/budgie-desktop#1019]
As I pointed out on G+, Budgie Settings currently doesn't work for me. I have no idea why as I can't get any output because it crashes the whole desktop.

1 I'm launching Budgie Settings from either the menu or CLI
2 Either Budgie restarts itself or I get thrown back to the login screen

So if you have any advice on how to debug this for you (as I said: because of the crash, I get no output)... I hope we can solve this :)

OS: Solus (latest, all updates installed)
Budgie: latest Git Master
(not sure if needed but anyway...) Video card: Intel Broadwell (integrated)

**Comment by ikeydoherty** [comment]
After logging in, run:

`nohup budgie-panel --replace&`

This'll stop the session restarting when Budgie Panel crashes. Now open the Settings UI and paste the output here when it crashes.

**Comment by Vistaus** [comment]
Thanks, that's a great trick, I'll try to remember that one!

Anyway, still no output :(

```
[12:27:29] vistaus@vistaus-solus /media/MicroSD/Apps (0) 
> nohup budgie-panel --replace&
nohup: ignoring input and appending output to 'nohup.out'
[12:27:30] vistaus@vistaus-solus /media/MicroSD/Apps (0) 
> budgie-desktop-settings
[12:27:39] vistaus@vistaus-solus /media/MicroSD/Apps (0) 
> fish: 'nohup budgie-panel --replace&' terminated by signal SIGTRAP (Trace or breakpoint trap)

[12:27:50] vistaus@vistaus-solus /media/MicroSD/Apps (0) 
> 
```

The panel did crash though.

**Comment by ikeydoherty** [comment]
Dumb question but are you actually on the latest git master? Because we fixed a crippling crash on that a few days ago

**Comment by Vistaus** [comment]
Yes, I'm on the latest git master. I update twice a day.

**Comment by ikeydoherty** [comment]
Might be worth running it via gdb to figure out why its going kersplat

**Comment by Vistaus** [comment]
Good call!

I got *some* output now:

`[12:36:11] vistaus@vistaus-solus /media/MicroSD/Apps (0) 
> gdb budgie-desktop-settings
GNU gdb (GDB) 7.12.1
Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-solus-linux".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from budgie-desktop-settings...done.
(gdb) run
Starting program: /usr/bin/budgie-desktop-settings 
warning: Unable to find libthread_db matching inferior's thread library, thread debugging will not be available.
[New LWP 10103]
[New LWP 10104]
[LWP 10103 exited]
[LWP 10099 exited]
[Inferior 1 (process 10099) exited normally]
(gdb) 
`

**Comment by ikeydoherty** [comment]
Don't use `budgie-desktop-settings`, it's a proxy call into `budgie-panel`

```
sudo eopkg it glibc-dbginfo libgtk-3-dbginfo
gdb budgie-panel

(gdb) run --replace
(gdb) bt
```

**Comment by Vistaus** [comment]
Ah, I see. Executed all of your commands and got this output:

```
[12:38:49] vistaus@vistaus-solus /media/MicroSD/Apps (0) 
> sudo eopkg it glibc-dbginfo libgtk-3-dbginfo
Password: 
Following packages will be installed:
glibc-dbginfo  libgtk-3-dbginfo
Total size of package(s): 7.46 MB
Downloading 1 / 2
Package glibc-dbginfo found in repository SolusUnstable
glibc-dbginfo-2.25-44-1-x86_64.eopkg (7.2 MB)100%      5.00 MB/s [00:00:01] [complete]
Downloading 2 / 2
Package libgtk-3-dbginfo found in repository SolusUnstable
libgtk-3-dbginfo-3.22.16-61-1-x86_64.eopkg (224.0 KB)100%    182.04 MB/s [00:00:00] [complete]
Installing 1 / 2
glibc-dbginfo-2.25-44-1-x86_64.eopkg [cached]
Installing glibc-dbginfo, version 2.25, release 44
Extracting the files of glibc-dbginfo
Configuring glibc-dbginfo package
Configured glibc-dbginfo
Installed glibc-dbginfo
Installing 2 / 2
libgtk-3-dbginfo-3.22.16-61-1-x86_64.eopkg [cached]
Installing libgtk-3-dbginfo, version 3.22.16, release 61
Extracting the files of libgtk-3-dbginfo
Configuring libgtk-3-dbginfo package
Configured libgtk-3-dbginfo
Installed libgtk-3-dbginfo
[12:39:00] vistaus@vistaus-solus /media/MicroSD/Apps (0) 
> gdb budgie-panel
GNU gdb (GDB) 7.12.1
Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-solus-linux".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from budgie-panel...done.
(gdb) run --replace
Starting program: /usr/bin/budgie-panel --replace
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib64/libthread_db.so.1".
[New Thread 0x7fffe7ece700 (LWP 10234)]
[New Thread 0x7fffe76cd700 (LWP 10235)]

(budgie-panel:10230): Gtk-WARNING **: Theme parsing error: gtk.css:2:0: Expected semicolon

(budgie-panel:10230): Gtk-WARNING **: Theme parsing error: gtk.css:3:0: Expected semicolon

(budgie-panel:10230): Gtk-WARNING **: Theme parsing error: gtk.css:4:0: Expected semicolon

(budgie-panel:10230): Gtk-WARNING **: Theme parsing error: gtk.css:5:0: Expected semicolon

(budgie-panel:10230): Gtk-WARNING **: Theme parsing error: gtk.css:3053:56: '/*' in comment block
[New Thread 0x7fffe69b5700 (LWP 10236)]
** Message: manager.vala:1392: Replaced existing budgie-panel
[New Thread 0x7fffe61b4700 (LWP 10238)]

(budgie-panel:10230): Gtk-WARNING **: Theme parsing error: gtk.css:3053:56: '/*' in comment block
** Message: IconTasklistApplet.vala:418: Invalid application! gnome-calculator.desktop

(budgie-panel:10230): Gtk-WARNING **: Theme parsing error: gtk.css:3053:56: '/*' in comment block
** Message: IconTasklistApplet.vala:418: Invalid application! gnome-calculator.desktop
[New Thread 0x7fffc33c7700 (LWP 10240)]
[Thread 0x7fffc33c7700 (LWP 10240) exited]
bt

[Thread 0x7fffe69b5700 (LWP 10236) exited]
^C
Thread 1 "budgie-panel" received signal SIGINT, Interrupt.
0x00007ffff61cbbbd in poll () at ../sysdeps/unix/syscall-template.S:84
84	../sysdeps/unix/syscall-template.S: No such file or directory.
(gdb) bt
#0  0x00007ffff61cbbbd in poll () at ../sysdeps/unix/syscall-template.S:84
#1  0x00007ffff6aa9da9 in  () at /usr/lib/libglib-2.0.so.0
#2  0x00007ffff6aaa162 in g_main_loop_run () at /usr/lib/libglib-2.0.so.0
#3  0x00007ffff73c0c0d in gtk_main () at /usr/lib/libgtk-3.so.0
#4  0x0000000000410737 in _vala_main (args=<optimized out>, args_length1=<optimized out>) at src/panel/budgie-panel@exe/main.c:121
#5  0x00007ffff60e74da in __libc_start_main (main=
    0x4103f0 <main>, argc=2, argv=0x7fffffffdcc8, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffdcb8) at ../csu/libc-start.c:295
#6  0x000000000041043a in _start () at ../sysdeps/x86_64/start.S:120
(gdb) 
```

**Comment by ikeydoherty** [comment]
Wtf. OK what if you do a new build of Budgie using `--buildtype debug` ? Output might be more useful

**Comment by Vistaus** [comment]
Alright, I've rebuild it with --buildtype debug, rebooted my laptop and this is the output I'm getting now:
```

[13:08:59] vistaus@vistaus-solus /home/vistaus (0) 
> gdb budgie-panel
GNU gdb (GDB) 7.12.1
Copyright (C) 2017 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-solus-linux".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from budgie-panel...done.
(gdb) run --replace
Starting program: /usr/bin/budgie-panel --replace
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib64/libthread_db.so.1".
[New Thread 0x7fffe7ece700 (LWP 1320)]
[New Thread 0x7fffe76cd700 (LWP 1321)]

(budgie-panel:1316): Gtk-WARNING **: Theme parsing error: gtk.css:2:0: Expected semicolon

(budgie-panel:1316): Gtk-WARNING **: Theme parsing error: gtk.css:3:0: Expected semicolon

(budgie-panel:1316): Gtk-WARNING **: Theme parsing error: gtk.css:4:0: Expected semicolon

(budgie-panel:1316): Gtk-WARNING **: Theme parsing error: gtk.css:5:0: Expected semicolon

(budgie-panel:1316): Gtk-WARNING **: Theme parsing error: gtk.css:3053:56: '/*' in comment block
[New Thread 0x7fffe69b5700 (LWP 1322)]
[New Thread 0x7fffe61b4700 (LWP 1324)]

(budgie-panel:1316): Gtk-WARNING **: Theme parsing error: gtk.css:3053:56: '/*' in comment block
** Message: IconTasklistApplet.vala:418: Invalid application! gnome-calculator.desktop
** Message: IconTasklistApplet.vala:418: Invalid application! menulibre-ramme-instagram-client2.desktop
[New Thread 0x7fffc33c7700 (LWP 1325)]
[Thread 0x7fffc33c7700 (LWP 1325) exited]
^C
Thread 1 "budgie-panel" received signal SIGINT, Interrupt.
0x00007ffff61cbbbd in poll () at ../sysdeps/unix/syscall-template.S:84
84	../sysdeps/unix/syscall-template.S: No such file or directory.
(gdb) bt
#0  0x00007ffff61cbbbd in poll () at ../sysdeps/unix/syscall-template.S:84
#1  0x00007ffff6aa9da9 in  () at /usr/lib/libglib-2.0.so.0
#2  0x00007ffff6aaa162 in g_main_loop_run () at /usr/lib/libglib-2.0.so.0
#3  0x00007ffff73c0c0d in gtk_main () at /usr/lib/libgtk-3.so.0
#4  0x0000000000410737 in _vala_main (args=<optimized out>, args_length1=<optimized out>) at /media/MicroSD/Apps/budgie-desktop/src/panel/main.vala:46
#5  0x00007ffff60e74da in __libc_start_main (main=
    0x4103f0 <main>, argc=2, argv=0x7fffffffe2c8, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7fffffffe2b8) at ../csu/libc-start.c:295
#6  0x000000000041043a in _start () at ../sysdeps/x86_64/start.S:120
(gdb) 
```

**Comment by ikeydoherty** [comment]
The mentioned line 46:

```vala
    Gtk.main();
```

..That helps us not at all.

**Comment by Vistaus** [comment]
What do you mean?

**Comment by ikeydoherty** [comment]
I mean there is no usable debug information at all and the crash isn't happening inside Budgie code lol.

Try commenting out this block of code which might be suspect:

```vala
        Idle.add(()=> {
            this.load_themes();
            return false;
        });
```

In `src/panel/settings/settings_style.vala` (92-95) and rebuild

**Comment by Vistaus** [comment]
Hmm, I got the settings to work now after removing two pinned apps. But it only runs when I use the nohup command you provided earlier.

**Comment by ikeydoherty** [comment]
huh ok. Well what if you do a full settings reset? `nohup budgie-panel --reset --replace&`

**Comment by Vistaus** [comment]
Yeah, I dunno. Maybe one of the pinned apps had a invalid .desktop file or something.

A full settings reset seems to have helped.

**Comment by Vistaus** [comment]
I have no idea what initially caused this issue, but after doing a few of the things you said, it's fully working now w/o having to 'nohup' first. I do notice that it still crashes when Nautilus is not installed (it even drops an error about that). I get that Nautilus is part of the default Solus application set, but shouldn't BDS work w/o Nautilus as well?

**Comment by ikeydoherty** [comment]
Woot? xD

**Comment by t3n3t** [comment]
Same here, Arch Linux. I suppose it could possibly be caused by recent Gnome 3.28 update.

**Comment by fossfreedom** [comment]
@gersonjferreira and @t3n3t  - whoever maintains your packages will need to patch budgie-desktop to ensure the new libmutter and gnome-settings-daemon requirements are met for GNOME 3.28.

**Comment by gersonjferreira** [comment]
You mean remove the packages from gnome group that are not Budgie dependencies, I guess, right? Later today I'll try to remove one by one to see if I can find the culprit...

**Comment by gersonjferreira** [comment]
Sorry, but I'm not able (or patient enough) to remove the unnecessary gnome packages. Here in Antergos they are all linked to meta packages... it is a pain! I'll probably move back to Manjaro for now. And later I can try Arch.

**Comment by t3n3t** [comment]
@gersonjferreira Just dont use meta packages next time you install Arch, lol. Got to stick back to gnome for some time, i guess.

**Comment by gersonjferreira** [comment]
Anterogos installs the meta packages by default! Indeed I wouldn't every install meta packages by myself! This is my first attempt with Antergos. I've been happily using Manjaro for a while (as a former Debian user), but I was willing to try Arch and Antergos... and since everybody says that Antergos = Arch with an easier installation process... I'm trying. But already not happy! 

Between Antergos and Manjaro, I prefer Manjaro already. But I use my desktop to work (I'm a physics professor)... so I can't spend too much time fixing distribution issues. So I'll move back to Manjaro for now, and during my next vacation I'll try Arch.

BTW, this is not the only meta-issue with Antergos. Yesterday I've posted in their forum complaining that the default install comes with a huge number of hunspell dictionaries linked to antergos-commom-meta, which cannot be removed directly, since it also installs the DM, etc... I had to mark the packages as explicitly installed and remove the meta + useless dicts.

To summarize... it seems that Budgie is suffering from abusive meta packages in Antergos.

**Comment by t3n3t** [comment]
@gersonjferreira "antergos-commom-meta" - which is a strong proof against "Antergos = Arch with an easier installation process" :). Good luck.

**Comment by gersonjferreira** [comment]
@sakibccr, I can now confirm that the problem is with packages in the Gnome group. I've decided to give it a try and remove the meta-package by marking everything on it as explicit installed, and them removing everything that was unnecessary for Budgie. Now the Budgie Settings works again! Thanks for the tip!

So, the problem is some conflict with a Gnome package. But I cannot tell which one for sure.

**Comment by laymonage** [comment]
I'm having the same issue on a freshly installed Arch Linux.

**Comment by gersonjferreira** [comment]
There was an update to the budgie-desktop package today in Arch. It seems to be fixed now.

**Comment by laymonage** [comment]
Can confirm. Now it works on my installation as well.

**Comment by t3n3t** [comment]
Can confirm too.
Still the panel resize bug is annoying!!11 lol
